Sources/Frontend/Commands/ScanBehavior.swift

```diff
@@ -61,13 +61,21 @@ final class ScanBehavior {
 
         do {
             results = try block(project)
+
             let interval = logger.beginInterval("result:output")
-            let filteredResults = OutputDeclarationFilter().filter(results)
+            let filteredResults = try OutputDeclarationFilter().filter(results)
 
             if configuration.autoRemove {
                 try ScanResultRemover().remove(results: filteredResults)
             }
 
+            if !filteredResults.isEmpty, let baselinePath = configuration.writeBaseline {
```

Comment:
Do you definitely want to save the `filteredResults` here? - if you forbid reading from and writing to a baseline at the same time, then that should be fine, but if you do allow that (see my comment below) then I think that the unfiltered results is what should be saved.

The kind of workflow I'm thinking of here is

1. Runs periphery with a baseline, finds N new issues. Fixes some, but decided to add the others to the baseline
2. Runs periphery again, reading from the existing baseline, and also writing to it. Sees the unfixed issues being reported.
3. Runs periphery again, reading from the new baseline (possibly the same filename as before). Now sees no issues.

Reply:
I've changed the implementation to carry over the existing baseline and include the `filteredResults` results for the new baseline.